2024 01 19 gui deposit

mattyg opened the pull request:
Modals for 'Deposit' and 'Withdraw', opened by clicking buttons on Vault Detail page. Does not include actually making the deposit / withdrawal



Note this PR has a known issue where I'm treating the graphql type Vault as if it only supports a single token -- I need to modify the graphql queries to use tokenVault instead and update this page accordingly.

![image](https://github.com/rainlanguage/rain.orderbook/assets/159270/3dfd17ea-6ff1-4fa5-916d-9447fb49a375)

thedavidmeister commented on tauri-app/src/lib/utils/number.ts:
```diff
@@ -0,0 +1,3 @@
+export function isStringValidNumber(val: string) {
+  return /^\d+.?[\d]+$/.test(val);
```
can we get some test cases around this?

mattyg replied:
removed and replaced with [@imask/svelte](https://www.npmjs.com/package/@imask/svelte)

mattyg commented:
- replaced bespoke input mask implementation with 
- added "max" button to auto-fill the input with the max amount (currently only supported in Withdraw modal)
- added error message indicating that input is > max

thedavidmeister commented on tauri-app/src/lib/ModalVaultWithdraw.svelte:
```diff
@@ -0,0 +1,87 @@
+      bind:value={amount}
+      bind:valueRaw={amountRaw}
+      symbol={vault.token_vaults ? vault.token_vaults[0].token.symbol : ''}
+      decimals={vault.token_vaults ? vault.token_vaults[0].token.decimals : 16}
```
16?

thedavidmeister replied:
actually i think we should fallback to 0 with a clear error/warning that shows to the user that they have to put the raw absolute value in because we _can't_ infer the decimals for them

mattyg replied:
it's resolved in PR #111 -- removed all the conditionals of vault.token_vaults
